File: Python3/프로그래머스/3/12927. 야근 지수/야근 지수.py
import heapq

def solution(n, works):
    works = [-work for work in works]
    heapq.heapify(works)
    a = 0
    while 0 < n and works:
        heapq.heappush(works, heapq.heappop(works) + 1)
        # heapq.heapify(works)
        n -= 1
    return sum(work ** 2 for work in works if work < 0)

File: C++/프로그래머스/3/12927. 야근 지수/야근 지수.cpp
#include <queue>
#include <vector>

using namespace std;

long long solution(int n, vector<int> works) {
    priority_queue<long long> pq;
    for(auto& work : works) {
        pq.push(work);
    }
    
    while(0 < n--) {
        if(pq.top() == 0) break;
        pq.push(pq.top() - 1);
        pq.pop();
    }
    
    long long answer = 0;
    
    while(!pq.empty()) {
        answer += pq.top() * pq.top();
        pq.pop();
    }
    
    return answer;
}